adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBInterstitialRenderer.m

```diff
@@ -0,0 +1,109 @@
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationInterstitialAdEventDelegate> _delegate;
+}
+
+- (void)renderInterstitialForAdConfiguration:
+(nonnull GADMediationInterstitialAdConfiguration *)adConfiguration
+                           completionHandler:(nonnull GADMediationInterstitialLoadCompletionHandler)completionHandler {
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block  GADMediationInterstitialLoadCompletionHandler originalCompletionHandler = [completionHandler copy];
+    _loadCompletionHandler = ^id<GADMediationInterstitialAdEventDelegate>(_Nullable id<GADMediationInterstitialAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterPanglePlacementID] ?: @"";\
+    if (!placementId.length) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorInvalidServerParameters, [NSString stringWithFormat:@"%@ cannot be nil.",GADMAdapterPanglePlacementID]);
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    _fullScreenAdVideo = [[BUFullscreenVideoAd alloc] initWithSlotID:placementId];
+    _fullScreenAdVideo.delegate = self;
+    [_fullScreenAdVideo setAdMarkup:adConfiguration.bidResponse];
+}
+
+#pragma mark - GADMediationInterstitialAd
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    [_fullScreenAdVideo showAdFromRootViewController:viewController];
+}
+
+#pragma mark -  BUFullscreenVideoAdDelegate
+- (void)fullscreenVideoMaterialMetaAdDidLoad:(BUFullscreenVideoAd *)fullscreenVideoAd {
+    if (_loadCompletionHandler) {
+        id<GADMediationInterstitialAdEventDelegate> delegate = _delegate;
+        delegate = _loadCompletionHandler(self,nil);
+    }
+}
+
+- (void)fullscreenVideoAd:(BUFullscreenVideoAd *)fullscreenVideoAd didFailWithError:(NSError *_Nullable)error {
+    if (_loadCompletionHandler) {
+        _loadCompletionHandler(nil, error);
+    }
+}
+
+- (void)fullscreenVideoAdWillVisible:(BUFullscreenVideoAd *)fullscreenVideoAd {
+    id<GADMediationInterstitialAdEventDelegate> delegate = _delegate;
+    [delegate willPresentFullScreenView];
+    [delegate reportImpression];
+}
+
+- (void)fullscreenVideoAdDidClick:(BUFullscreenVideoAd *)fullscreenVideoAd {
+    id<GADMediationInterstitialAdEventDelegate> delegate = _delegate;
+    [delegate reportClick];
+}
+
+- (void)fullscreenVideoAdWillClose:(BUFullscreenVideoAd *)fullscreenVideoAd {
+    id<GADMediationInterstitialAdEventDelegate> delegate = _delegate;
+    [delegate willDismissFullScreenView];
```

Comment:
Ditto, can you double check if this will be invoked?  I couldn't see that on my end.